src/arrayMethodSort.js

```diff
@@ -4,8 +4,43 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let len = this.length;
+    let swapped;
+
+    do {
+      swapped = false;
+
+      for (let i = 0; i < len - 1; i++) {
+        if (compareFunction) {
+          if (compareFunction(this[i], this[i + 1]) > 0) {
+            const temp = this[i];
+
+            this[i] = this[i + 1];
+            this[i + 1] = temp;
+
+            swapped = true;
+          } else if (
+            compareFunction(this[i], this[i + 1]) < 0 ||
+            compareFunction(this[i], this[i + 1]).isNan()
+          ) {
+            swapped = true;
+          }
+        } else {
+          if (this[i].toString() > this[i + 1].toString()) {
+            const temp = this[i];
+
+            this[i] = this[i + 1];
+            this[i + 1] = temp;
+            swapped = true;
+          }
+        }
+      }
+
+      len--;
```

Comment:
You are using a bubble sort algorithm, which is fine for an educational exercise. However, be aware that bubble sort has a time complexity of O(n^2) which is not efficient for larger arrays. For production code, more efficient algorithms like quicksort or mergesort are generally used.